src/uct/ib/rc/accel/rc_mlx5_iface.c

```diff
@@ -197,31 +197,29 @@ uct_rc_mlx5_iface_handle_failure(uct_ib_iface_t *ib_iface, void *arg,
                                                                        qp_num),
                                                 uct_rc_mlx5_ep_t);
     ucs_log_level_t    log_lvl = UCS_LOG_LEVEL_FATAL;
-    uct_ib_mlx5_txwq_t txwq_copy;
-    size_t             txwq_size;
+    uint16_t           sn      = ntohs(cqe->wqe_counter);
 
     if (!ep) {
         return;
     }
 
-    /* Create a copy of RC txwq for completion error reporting, since the QP
-     * would be released by set_ep_failed()*/
-    txwq_copy = ep->tx.wq;
-    txwq_size = UCS_PTR_BYTE_DIFF(ep->tx.wq.qstart, ep->tx.wq.qend);
-    txwq_copy.qstart = ucs_malloc(txwq_size, "rc_txwq_copy");
-    if (txwq_copy.qstart != NULL) {
-        memcpy(txwq_copy.qstart, ep->tx.wq.qstart, txwq_size);
-        txwq_copy.qend = UCS_PTR_BYTE_OFFSET(txwq_copy.qstart, txwq_size);
+    uct_rc_mlx5_common_update_tx_res(iface, &ep->tx.wq, &ep->super.txqp, sn);
+    ep->super.fc.fc_wnd++;
```

Comment:
- when getting 1st error, set FC WND to MAX for DC as well
- check error before posting NOP from flush